File: scheduler-security/src/main/java/com/averude/uksatse/scheduler/security/entity/UserAccount.java
package com.averude.uksatse.scheduler.security.entity;

import lombok.Data;
import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.util.StringJoiner;

@Data
@Cacheable
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
@Entity(name = "user_accounts")
@Inheritance(strategy = InheritanceType.JOINED)
public class UserAccount implements Serializable {
    @Id
    private String username;
    @NotNull
    private String password;
    @NotNull
    private boolean locked = false;
    @NotNull
    private boolean enabled = true;

    @Override
    public String toString() {
        return new StringJoiner(", ", UserAccount.class.getSimpleName() + "[", "]")
                .add("username='" + username + "'")
                .toString();
    }
}

File: frontend-server/src/app/model/accounts/user-account.ts
import { IdEntity } from "../interface/id-entity";

export class UserAccount implements IdEntity{
  id:           number;
  username:     string;
  name:         string;
  role:         string;
  authority:    string;
  enterpriseId: number;
  departmentId: number;
  locked:       boolean;
  enabled:      boolean;
}
